### PR TITLE
feat: Textarea and TextLabel components

### PR DESCRIPTION
**Pull request by cheton**
* Added `Textarea` component
* Renamed `InputLabel` to `TextLabel`

### CODE REVIEW

**Comment by cheton** [comment]
# Textarea

The `Textarea` component allows you to create a multi-line text input.

## Import

```js
import Textarea from '@trendmicro/react-styled-core/Textarea';
// or
import { Textarea } from '@trendmicro/react-styled-core';
```

## Usage

```jsx
<>
  <TextLabel mb="1x">Label:</TextLabel>
  <Textarea placeholder="Basic example" />
  <Text size="xs" mt="1x">Help text for the text input</Text>
</>
```

### Sizing

* The `rows` and `cols` properties allow you to specify an exact size for the `<Textarea>` to take. Setting these is a good idea for consistency, as browser defaults can differ.
* The `maxLength` property specifies a maximum number of characters that the `Textarea` is allowed to contain.
* The `minLength` property specifies a minimum length that is considered valid. `Textarea` will not submit (and is invalid) if it is empty, using the `required` attribute.
* The `resize` property to set whether the `Textarea` is resizable, and if so, in which directions. You can set the value to `none`, `both`, `horizontal`, or `vertical`.

```jsx

function Example() {
  const [resize, setResize] = React.useState('both');

  return (
    <>
      <Text mb="2x">resize:</Text>
      <Stack direction="row" spacing=".5rem" mb="4x">
        <Button variant="outline" selected={resize === 'none'} onClick={() => setResize('none')}>none</Button>
        <Button variant="outline" selected={resize === 'both'} onClick={() => setResize('both')}>both</Button>
        <Button variant="outline" selected={resize === 'horizontal'} onClick={() => setResize('horizontal')}>horizontal</Button>
        <Button variant="outline" selected={resize === 'vertical'} onClick={() => setResize('vertical')}>vertical</Button>
      </Stack>
      <Textarea
        defaultValue="Placeholder text"
        width="auto"
        resize={resize}
        rows="3"
        cols="12"
        minLength={4}
        maxLength={16}
      />
    </>
  );
}
```

### Variants

The `Textarea` component comes in two variants: `outline`, `filled`, and `unstyled`. Pass the `variant` prop and set it to either of these values.

#### `outline`

```jsx
<Textarea variant="outline" placeholder="John Doe" />
```

#### `filled`

```jsx
<Textarea variant="filled" placeholder="John Doe" />
```

#### `unstyled`

```jsx
<Textarea variant="unstyled" placeholder="John Doe" />
```

### Attributes

Standard textarea attributes are supported, e.g., `disabled`, `readOnly`, `required`, etc.

#### `disabled`

```jsx
<Stack directin="column" spacing="4x">
  <Textarea disabled placeholder="Placeholder text" />
  <Textarea disabled placeholder="Placeholder text" defaultValue="Disabled" />
</Stack>
```

#### `readOnly`

```jsx
<Stack directin="column" spacing="4x">
  <Textarea readOnly placeholder="Placeholder text" />
  <Textarea readOnly placeholder="Placeholder text" defaultValue="Read-only" />
</Stack>
```

#### `required`

```jsx
<Textarea required placeholder="Placeholder text" />
```

### Validation

Use the `aria-invalid` attribute to indicate that the value entered into an input field does not conform to the format expected by the application. `aria-invalid` can also be used to indicate that a required field has not been filled in.

Set `aria-invalid` to `true` on the fields that have failed validation, otherwise set it to `false` if no errors detected.

```jsx noInline
const InlineError = (props) => (
  <Text fontSize="sm" lineHeight="sm" color="red:50" {...props} />
);

const MultilineInputField = ({
  value,
  onChange,
  ...props
}) => {
  const isInvalid = (value === '');
  let invalidTextareaStyle = {};
  if (isInvalid) {
    invalidTextareaStyle = {
      'aria-invalid': true,
      pr: '10x',
    };
  }

  return (
    <>
      <Flex position="relative" alignItems="center" mb="1x">
        <Textarea
          resize="none"
          rows="3"
          placeholder="Placeholder text"
          value={value}
          onChange={onChange}
          {...invalidTextareaStyle}
        />
        {isInvalid && (
          <Box position="absolute" right={0} top=".375rem">
            <TMIcon name="warning-circle" px="3x" color="red:50" />
          </Box>
        )}
      </Flex>
      <Box>
        {isInvalid && (
          <InlineError>This is a required field.</InlineError>
        )}
      </Box>
    </>
  );
};

function Example() {
  const [value, setValue] = React.useState('');
  const onChange = (e) => {
    setValue(e.target.value);
  };

  return (
    <MultilineInputField value={value} onChange={onChange} />
  );
}

render(<Example />);
```

## Props

`Textarea` composes the [`InputBase`](./inputbase) component.

| Name | Type | Default | Description |
| :--- | :--- | :------ | :---------- |
| `rows` | `number` | | Specifies the visible number of lines in a text area. |
| `cols` | `number` | | Specifies the visible width of a text area. |
| `maxLength` | `number` | | Specifies a maximum number of characters that is allowed to contain. |
| `minLength` | `number` | | Specifies a minimum length that is considered valid. |
| `resize` | `string` | | One of: 'none', 'both', 'horizontal', 'vertical' |
| `variant` | `string` | 'outline' | The variant of the input style to use. One of: 'outline', 'filled', 'unstyled' |
| `disabled` | `boolean` | | If `true`, the input will be disabled. This sets `aria-disabled=true` and you can style this state by passing the `_disabled` prop. |
| `readOnly` | `boolean` | | If `true`, prevents the value of the input from being edited. |
| `aria-invalid` | `boolean` | | If `true`, the input will indicate an error. You can style this state by passing the `_invalid` prop. |

**Comment by cheton** [comment]
# TextLabel

The `TextLabel` component is a label that composes the `Text` component.

## Import

```js
import TextLabel from '@trendmicro/react-styled-core/TextLabel';
// or
import { TextLabel } from '@trendmicro/react-styled-core';
```

## Usage

```jsx
<TextLabel>Label text:</TextLabel>
```

The text colors in the dark mode and light mode are `white:secondary` and `black:secondary`, respectively, defined in `theme.colors`.

## Props

`TextLabel` composes the [`Text`](./text) component. You can override the default styles with style props.

| Name | Type | Default | Description |
| :--- | :--- | :------ | :---------- |
| `size` | `string` | | One of: '4xl', '3xl', '2xl', 'xl', 'lg', 'md', 'sm', 'xs' |

File: packages/react-styled-core/src/Textarea/styles.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+};
+
+const sizes = {
+  'md': {
+    borderRadius: 'sm',
+    fontSize: 'sm',
+    lineHeight: 'sm',
+    minHeight: '8x',
+    px: '3x',
+    py: '.3125rem',
```
Comment (tinaClin) [inline packages/react-styled-core/src/Textarea/styles.js]: padding-top和padding-bottom和VD不一樣
image

File: packages/react-styled-core/src/Textarea/styles.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  return {
+    border: 1,
+    borderColor,
+    color,
+    _hover: {
+      borderColor: hoverBorderColor,
+
+      // Use a higher z-index value to bring overlapping border to front when hovered
+      zIndex: 2,
```
Comment (tinaClin) [inline packages/react-styled-core/src/Textarea/styles.js]: 設`z-index`是`Textarea`有重疊的情況？

File: packages/react-styled-core/src/Textarea/styles.js (hunk 1 above)
Comment (cheton) [reply packages/react-styled-core/src/Textarea/styles.js]: px 我看一下，至於 py 是 5px + 1px 沒錯

File: packages/react-styled-core/src/Textarea/styles.js (hunk 2 above)
Comment (cheton) [reply packages/react-styled-core/src/Textarea/styles.js]: You're right. `z-index` is not necessary for the textarea.

File: packages/react-styled-core/src/Textarea/styles.js (hunk 1 above)
Comment (cheton) [reply packages/react-styled-core/src/Textarea/styles.js]: 蛤!!  pb 是 10px

File: packages/react-styled-core/src/Textarea/styles.js (hunk 2 above)
Comment (cheton) [reply packages/react-styled-core/src/Textarea/styles.js]: fixed in f76b2bb1b57989210cf5fd1253d920a9957f910e

File: packages/react-styled-core/src/Textarea/styles.js (hunk 1 above)
Comment (cheton) [reply packages/react-styled-core/src/Textarea/styles.js]: @tinaClin fixed in cb60f55b36e964dcad6817b1603fde2f3b7628a7 and 8d58b367e4972e5982040d0ffcf697aa4baf0d26

File: packages/react-styled-core/src/Input/styles.js [hunk 3]
```diff
@@ -128,34 +130,51 @@ const getUnstyledStyle = ({
...
+  const variantProps = getVariantProps(_props);
+
+  return {
+    ...baseProps,
+    ...sizeProps,
+    ...variantProps,
+  };
+};
+
 export {
```
Comment (tinaClin) [inline packages/react-styled-core/src/Input/styles.js:199]: 如果是要跟其他component一致的話，請`export default useInputStyle;` XD

File: packages/react-styled-core/src/InputGroupAddon/styles.js [hunk 4]
```diff
@@ -79,34 +81,51 @@ const getUnstyledStyle = ({
...
+  const variantProps = getVariantProps(_props);
+
+  return {
+    ...baseProps,
+    ...sizeProps,
+    ...variantProps,
+  };
+};
+
 export {
```
Comment (tinaClin) [inline packages/react-styled-core/src/InputGroupAddon/styles.js:129]: 如果是要跟其他component一致的話，請`export default useInputGroupAddonStyle;` XD

Comment (cheton) [reply packages/react-styled-core/src/InputGroupAddon/styles.js:129]: No! 除非檔名是 `useInputGroupAddonStyle.js`

File: packages/react-styled-core/src/Input/styles.js (hunk 3 above)
Comment (cheton) [reply packages/react-styled-core/src/Input/styles.js:199]: No! 除非檔名是 `useInputStyle.js`

Comment (cheton) [reply packages/react-styled-core/src/Input/styles.js:199]: 這裡比較像是 utility function 的 export，至於 export default 最好是跟檔名一樣性質的東西，Chakra 那樣是不好的示範

**Review by tinaClin — APPROVED** [review]

File: packages/react-styled-core/src/Input/index.js [hunk 5]
```diff
@@ -56,7 +38,10 @@ const Input = forwardRef((
     <InputBase
       ref={ref}
       as="input"
-      css={css}
+      css={[ // TODO: combine the css array in v11
+        getInputGroupCSS({ variant }),
+        css,
+      ]}
```
Comment (cheton) [inline packages/react-styled-core/src/Input/index.js:44]: @tinaClin I finally decided to use this name within the `Input` component. I also found it's not necessary to use spread operator here for object expansion. It was handled by the `emotion` internally.

**Pull request merged**